tests/test_orient_symmetric.py

```diff
@@ -1,28 +1,198 @@
+from aspire.volume import Volume
 
 
 class OrientSymmTestCase(TestCase):
+    # `order` is at this scope to be picked up by parameterized in testCommonLines.
+    order = 3
+
     def setUp(self):
-        self.L = 32
-        self.symm = "C4"
+        self.L = 64
         self.n_img = 32
-        self.seed = 8675309
-        src = Simulation(L=self.L, n=self.n_img, symmetry=self.symm)
-        self.cl_class = CLSymmetryC3C4(
-            src, symmetry=self.symm, n_theta=360, seed=self.seed
-        )
+        self.dtype = np.float32
+        self.n_theta = 360
+
+        # Build symmetric volume and associated Simulation object and common-lines class.
+        # For the Simulation object we use clean, non-shifted projection images.
+        orders = [3, 4]
+        self.vols = {}
+        self.srcs = {}
+        self.cl_classes = {}
+
+        for o in orders:
+            self.vols[o] = self.buildSimpleSymmetricVolume(self.L, o)
+
+            self.srcs[o] = Simulation(
+                L=self.L,
+                n=self.n_img,
+                offsets=np.zeros((self.n_img, 2)),
+                dtype=self.dtype,
+                vols=self.vols[o],
+                C=1,
+            )
+
+            self.cl_classes[o] = CLSymmetryC3C4(
+                self.srcs[o], symmetry=f"C{o}", n_theta=self.n_theta
+            )
 
     def tearDown(self):
         pass
 
+    @parameterized.expand([(order,), (order + 1,)])
+    def testRelativeRotations(self, order):
+        n_img = self.n_img
+
+        # Simulation source and common lines Class corresponding to
+        # volume with C3 or C4 symmetry.
+        src = self.srcs[order]
+        cl_symm = self.cl_classes[order]
+
+        # Estimate relative viewing directions.
+        cl_symm.build_clmatrix()
+        cl = cl_symm.clmatrix
+        Rijs = cl_symm._estimate_all_Rijs_c3_c4(cl)
+
+        # Each Rij belongs to the set {Ri.Tg_n^sRj, JRi.Tg_n^sRjJ},
+        # s = 1, 2, ..., order. We find the mean squared error over
+        # the minimum error between Rij and the above set.
+        rots_symm = self.buildCyclicRotations(order)
+        gs = rots_symm
+        J = np.diag([-1, -1, 1])
+        rots_gt = src.rots
+
+        nchoose2 = int(n_img * (n_img - 1) / 2)
+        min_idx = np.zeros(nchoose2, dtype=int)
+        errs = np.zeros(nchoose2)
+        diffs = np.zeros(order)
+        pairs = all_pairs(n_img)
+        for idx, (i, j) in enumerate(pairs):
+            Rij = Rijs[idx]
+            Ri_gt = rots_gt[i]
+            Rj_gt = rots_gt[j]
+            for s in range(order):
+                Rij_s_gt = Ri_gt.T @ gs[s] @ Rj_gt
+                diffs[s] = np.minimum(
+                    norm(Rij - Rij_s_gt), norm(J @ Rij @ J - Rij_s_gt)
+                )
+            min_idx[idx] = np.argmin(diffs)
+            errs[idx] = diffs[min_idx[idx]]
+
+        mse = np.mean(errs**2)
+
+        # Mean-squared-error is better for C3 than for C4.
```

Comment:
Why is that?

Reply:
I'm not really sure. I might have to tinker with this a bit. Here is a plot of the errors for C3: 
![image](https://user-images.githubusercontent.com/66811911/176461794-2e5807a0-baa2-4812-bd54-7e085bb36655.png)

There is basically one estimate that is way off. For C4 (below) there are about 10 bad estimates (out of 496 total estimates:![image](https://user-images.githubusercontent.com/66811911/176462161-85c9d390-d298-4e67-a791-fea875d017d7.png)

Reply:
Huh. Might be worth investigating closer. Can you make an issue?

Reply:
Created issue #663.